### PR TITLE
Mark Linux fallocate extensions as specific to Linux

### PR DESCRIPTION
**Pull request by mattmacy**
Signed-off-by: Matt Macy <mmacy@FreeBSD.org>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/os/linux/zfs/vdev_file.c [hunk 1]
```diff
@@ -268,10 +268,12 @@ vdev_file_io_start(zio_t *zio)
 		zio_execute(zio);
 		return;
 	} else if (zio->io_type == ZIO_TYPE_TRIM) {
-		int mode;
+		int mode = 0;
 
 		ASSERT3U(zio->io_size, !=, 0);
+#ifdef __linux__
```
Comment (behlendorf) [inline module/os/linux/zfs/vdev_file.c]: Is the `__linux__` needed here?  This file is already part of Linux platform code where we're allowed to do platform specific things.

Comment (mattmacy) [reply module/os/linux/zfs/vdev_file.c]: The FreeBSD user space will use the linux spl for libzpool

Comment (behlendorf) [reply module/os/linux/zfs/vdev_file.c]: I see, ok this looks good.

Comment (mattmacy) [reply module/os/linux/zfs/vdev_file.c]: This line is unchanged on FreeBSD:
https://github.com/zfsonlinux/zfs/blob/master/lib/libzpool/Makefile.am#L7

**Pull request merged**